### PR TITLE
On-ramp: Use dynamic list of networks

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR adopts a dynamic list of supported networks for the on-ramp aggregator.
- `on-ramp-sdk` is updated to version `1.12.0`, this version includes `getNetworks` method
- `allowedToBuy` method and hardcoded list was removed
-  `useOnRampNetworks` hook is added, it updates the networks on the state with the result of `getNetworks`
    - This update is done on mount and on network change
-  `useOnRampNetwork` hook and util functions are added to replace `allowedToBuy` 

**Screenshots/Recordings**

~~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by wachunei** [comment]
@bkirb can you confirm this is QAd please 🙏

**Comment by sethkfman** [comment]
@wachunei Can you resolve the conflicts on this PR?

**Comment by wachunei** [comment]
what conflicts? 😜  just did lol

File: app/components/UI/FiatOnRampAggregator/hooks/useOnRampNetworks.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  updateOnRampNetworks,
+} from '../../../../reducers/fiatOrders';
+import Logger from '../../../../util/Logger';
+
+/**
+ * Hook that fetches the available networks from the aggregator
+ * and updates the store
+ * @returns {null}
+ */
+function useOnRampNetworks() {
```
Comment (Cal-L) [inline app/components/UI/FiatOnRampAggregator/hooks/useOnRampNetworks.ts]: Nit - update name `useFetchOnRampNetworks`

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**